text/0000-partial_types.md

```diff
@@ -0,0 +1,672 @@
+    type UpdType;
+
+    fn summarize<%a>(&self: & %a Self) -> String;
+
+    fn update_value<%a>(&mut self : &mut %a Self, newvalue: UpdType);
+
+    fn update_sametype<%a, %b>(&mut self : &mut %a Self, &another: & %b Self);
+}
+```
+
+Unfortunately, having variants of type integrity is not enough to write **safe** implementations or other non-virtual function declarations.
+
+
+## Detailed Integrity
+
+We need detailed integrity to write non-virtual specific typed parameters in function, including trait implementation.
+
+An abstractions is added for integrity detailing, we assume that **every** variable is a `struct`-like objects (even if it is not).
+
+We need for this some new quasi-fields and some field integrity (which should be soft keywords).
+
+_Note: I do not comment types (as is needed for compiler) for colorizing purposes._
+
+_Note: I use symbol `~` as a synonym of "equivalent" word only. It is not a Rust operator._
+```rust
+// case (C1)
+let foo = 0i16;
+    foo : i16  ~  %full i16;
+    foo : %full i16  ~  %{self.*} i16;
+    foo : %{self.*} i16  ~  %{self.self} i16;
+```
+
+Where :
+ - `self` is an "access" to variable itself
+ - `.*` is an "every field" quasi-field
+ - `.self` quasi-field for primitive types, since `i16` is not a `struct` type
+
+What's about structures? Almost the same:
+```rust
+struct Point {
+    x: f64,
+    y: f64,
+    z: f64,
+    t: f64,
+    w: f64,
+}
+
+// case (C2)
+let &mut p1 : &mut Point = Point {x:1.0, y:2.0, z:3.0, t:4.0, w:5.0};
+    p1 : &mut Point  ~  &mut %full Point;
+    p1 : &mut %{self.*} Point;
+    p1 : &mut %{self.x, self.y, self.z, self.t, self.w} Point;
+    p1 : &mut %{self.{x, y, z, w}} Point;
+```
+
+Where :
+ - `.{<fld1>, <fld2>, }` is an field-set quasi-field
+
+We assume, that each field could be in one of 2 specific field-integrity - `%fit` and `%deny`. 
+
+We also must reserve as a keyword a `%miss` field-integrity for future ReExtendeded Partial Types, which allows to create **safe** self-referential types.
+
+`%fit` is default field-integrity and it means we have an access to this field and could use it as we wish. But if we try to access `%deny` field it cause a compiler error.
+
+```rust
+// case (C3)
+let foo = 0i16;
+    foo : %{self.*} i16  ~  %{self.self} i16;
+    foo : %{%fit self.*} i16  ~  %{%fit self.self} i16;
+    foo : %{%fit self.*, %deny self._} i16;
+
+
+// case (C4)
+// FROM case (C2)
+let &mut p1 : &mut Point = Point {x:1.0, y:2.0, z:3.0, t:4.0, w:5.0};
+    p1 : &mut %{%fit self.*} Point  ~  &mut  %{%fit self.*, %deny self._};
+    p1 : &mut %{%fit self.{x, y, z, w}} Point;
+```
+
+Where :
+ - `%fit` integrity
+ - `%deny` integrity
+ - `._` is a "rest of fields" quasi-field
+
+As we see, 
+ - `%empty : %{%deny self.*}` or `%empty : %{}` integrity
+ - `%full  : %{%fit  self.*}` or `%full  : %{self.*}` integrity
+
+
+## Partial parameters
+
+We add enough integrity, and could write partial parameters for non-virtual function declarations:
```

Comment:
Since this doesn't involve traits, I don't see how virtual functions even apply here.